zkevm-circuits/src/tx_circuit.rs

```diff
@@ -1690,6 +1715,138 @@ impl<F: Field> SubCircuitConfig<F> for TxCircuitConfig<F> {
             },
         );
 
+        //////////////////////////////////////////////////////////
+        //// EIP4844: Accumulation and Hashing of Chunk Bytes  ///
+        //////////////////////////////////////////////////////////
+        meta.create_gate("Chunk Bytes RLC", |meta| {
+            let mut cb = BaseConstraintBuilder::default();
+
+            // Accumulate hash length
+            cb.require_equal(
+                "chunk_txbytes_len_acc::cur == chunk_txbytes_len_acc::prev + HashLength",
+                meta.query_advice(chunk_txbytes_len_acc, Rotation::cur()),
+                meta.query_advice(chunk_txbytes_len_acc, Rotation(-(HASH_RLC_OFFSET as i32)))
```

Comment:
How does this behave for the first occurrence of `TxHashRlc` tag? You would be looking -20 relative offset, that can be assigned non-zero by a malicious prover?

Reply:
You might need to constrain the `len_acc` and `bytes_rlc` to equal `0` at `q_first` fixed column.

Reply:
Also, from the max number (that is fixed) of txs that we assign, a few towards the end could be padding. So we need to also account for:
```python
if tx.is_padding:
    # len_acc does not change
    # bytes_rlc does not change
```

Reply:
`q_first` conditions added. padding txs excluded and relevant constraints added.